src/connectionController.ts

```diff
@@ -69,20 +71,24 @@ interface ConnectionQuickPicks {
   data: { type: NewConnectionType; connectionId?: string };
 }
 
-interface ConnectionSecretsInfo {
-  connectionId: string;
-  secrets: ConnectionSecrets;
-}
+type StoreConnectionInfoWithConnectionModel = StoreConnectionInfo &
+  Required<Pick<StoreConnectionInfo, 'connectionModel'>>;
 
 type StoreConnectionInfoWithConnectionOptions = StoreConnectionInfo &
   Required<Pick<StoreConnectionInfo, 'connectionOptions'>>;
 
+type MigratedStoreConnectionInfo = StoreConnectionInfo &
+  Required<Pick<StoreConnectionInfo, 'secretStorageLocation'>>;
+
+type MigratedStoreConnectionInfoWithConnectionOptions =
+  StoreConnectionInfoWithConnectionOptions & MigratedStoreConnectionInfo;
+
 export default class ConnectionController {
   // This is a map of connection ids to their configurations.
   // These connections can be saved on the session (runtime),
   // on the workspace, or globally in vscode.
   _connections: {
-    [connectionId: string]: StoreConnectionInfoWithConnectionOptions;
+    [connectionId: string]: MigratedStoreConnectionInfoWithConnectionOptions;
   } = {};
```

Comment:
Just something I noticed while reviewing:

```suggestion
  } = Object.create(null);
```

Shouldn’t be an issue in practice, but this is just good practice for objects with generic keys

Reply:
TIL and thank you for this 😄
I always used to do `for in + if hasOwnProperty` or `Object.keys`, not anymore.

Reply:
`Object.keys()` is still a good thing to use, I think ;)